formattable is broken on MSVC

Does not compile

```
D:\a\strong_type\strong_type\include\strong_type/strong_type.hpp(1636,1): error C2977: 'std::formatter': too many template arguments [D:\a\strong_type\strong_type\build\self_test.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.32.31326\include\format(2865): message : see declaration of 'std::formatter' [D:\a\strong_type\strong_type\build\self_test.vcxproj]
D:\a\strong_type\strong_type\include\strong_type/strong_type.hpp(1657,2): error C2977: 'std::formatter': too many template arguments [D:\a\strong_type\strong_type\build\self_test.vcxproj]
D:\a\strong_type\strong_type\include\strong_type/strong_type.hpp(1636): message : see declaration of 'std::formatter' [D:\a\strong_type\strong_type\build\self_test.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.32.31326\include\chrono(6347,69): error C2977: 'std::formatter': too many template arguments [D:\a\strong_type\strong_type\build\self_test.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.32.31326\include\chrono(6346): message : see declaration of 'std::formatter' [D:\a\strong_type\strong_type\build\self_test.vcxproj]
```

--- a/include/strong_type/strong_type.hpp
+++ b/include/strong_type/strong_type.hpp
@@ -33,11 +33,16 @@
 #endif
 
 #ifndef STRONG_HAS_STD_FORMAT
-#if __has_include(<format>)
+#if __has_include(<version>)
+#include <version>
+#if defined(__cpp_lib_format) && __cpp_lib_format >= 201907
 #define STRONG_HAS_STD_FORMAT 1
-#else
-#define STRONG_HAS_STD_FORMAT 0
 #endif
+#endif
+#endif
+
+#ifndef STRONG_HAS_STD_FORMAT
+#define STRONG_HAS_STD_FORMAT 0
 #endif
 
 #ifndef STRONG_HAS_FMT_FORMAT
@@ -50,10 +55,6 @@
 
 #if STRONG_HAS_STD_FORMAT
 #include <format>
-#if !defined(__cpp_lib_format) || __cpp_lib_format < 201907
-#undef STRONG_HAS_STD_FORMAT
-#define STRONG_HAS_STD_FORMAT 0
-#endif
 #endif
 
 #if STRONG_HAS_FMT_FORMAT

--- a/include/strong_type/strong_type.hpp
+++ b/include/strong_type/strong_type.hpp
@@ -1633,16 +1633,16 @@ struct is_arithmetic<::strong::type<T, Tag, M...>>
 
 #if STRONG_HAS_STD_FORMAT
 template<typename T, typename Tag, typename... M, typename Char>
-struct formatter<::strong::type<T, Tag, M...>, Char,
-                 std::enable_if_t<
-                     std::is_base_of<
-                         ::strong::formattable::modifier<
-                             ::strong::type<T, Tag, M...>
-                             >,
-                         ::strong::type<T, Tag, M...>
-                         >::value
-                     >>
-    : formatter<T>
+struct formatter<::strong::type<T, Tag, M...>, Char>
+  : std::enable_if_t<
+        std::is_base_of<
+            ::strong::formattable::modifier<
+                ::strong::type<T, Tag, M...>
+            >,
+            ::strong::type<T, Tag, M...>
+        >::value,
+        formatter<T>
+    >
 {
   using type = ::strong::type<T, Tag, M...>;
   template<typename FormatContext>
@@ -1663,16 +1663,16 @@ struct formatter<::strong::type<T, Tag, M...>, Char,
 namespace fmt
 {
 template<typename T, typename Tag, typename... M, typename Char>
-struct formatter<::strong::type<T, Tag, M...>, Char,
-                 std::enable_if_t<
-                   std::is_base_of<
-                     ::strong::formattable::modifier<
-                       ::strong::type<T, Tag, M...>
-                     >,
-                     ::strong::type<T, Tag, M...>
-                   >::value
-                 >>
-  : formatter<T>
+struct formatter<::strong::type<T, Tag, M...>, Char>
+  : std::enable_if_t<
+        std::is_base_of<
+            ::strong::formattable::modifier<
+                ::strong::type<T, Tag, M...>
+            >,
+            ::strong::type<T, Tag, M...>
+        >::value,
+        formatter<T>
+     >
 {
   using type = ::strong::type<T, Tag, M...>;
   template<typename FormatContext>

--- a/test.cpp
+++ b/test.cpp
@@ -1323,7 +1323,11 @@ TEST_CASE("format")
 
 #if STRONG_HAS_FMT_FORMAT
   CHECK(fmt::format("{:d}", fi) == fmt::format("{:d}", 5));
+#if FMT_VERSION >= 80000
+  CHECK_THROWS_AS(fmt::format(fmt::runtime("{:s}"), fi), fmt::format_error);
+#else
   CHECK_THROWS_AS(fmt::format("{:s}", fi), fmt::format_error);
+#endif
 #endif
 
 #if STRONG_HAS_STD_FORMAT

--- a/test.cpp
+++ b/test.cpp
@@ -1328,7 +1328,8 @@ TEST_CASE("format")
 
 #if STRONG_HAS_STD_FORMAT
   CHECK(std::format("{:d}", fi) == std::format("{:d}", 5));
-  CHECK_THROWS_AS(std::format("{:s}", fi), std::format_error);
+  // Use std::vformat to mitigate the compile time check.
+  CHECK_THROWS_AS(std::vformat("{:s}", std::make_format_args(fi)), std::format_error);
 #endif
 }
 #endif

